### ISSUE TITLE
Storybook vuejs scss global scss helper file not available via webpack

### ISSUE THREAD

**Issue by cojok** [storybookjs/storybook#6743]
**Describe the bug**
There seems to be an issue in rendering the vue component, as I get alsway 'Undefined variable' for my scss variable from the style tag included in the vue file. This variable comes from a global helpers.scss, which is loaded in the vue app from webpack as loader scss data.

I have a vuejs setup done withe the vue-cli.  I have added storybook to it manually, done like in the tutorial from learnstorybook.com

The vue-cli-plugin-storybook did also not helped me.

**To Reproduce**
Project setup:
|src
|---|components
|---|---|Component1
|---|---|---|Component1.vue
|---|---|---|Component1.story.js
|---|sass
|---|---|utils
|---|---|---|utils1.scss
|---|---|---|utils2.scss
|---|---|helpers.scss

My components are vue single file components. I have also some webpack custom settings like aliases and the global loader of the helpers.scss, so I don't have to include it in every component.
2. run `$ npm run storybook`
3. see the error: 
```
ERROR in ./src/components/Error/Error.vue?vue&type=style&index=0&id=395d98a1&lang=scss&scoped=true& (./node_modules/css-loader!./node_modules/vue-loader/lib/loaders/stylePostLoader.js!./node_modules/sass-loader/lib/loader.js!./node_modules/vue-loader/lib??vue-loader-options!./src/components/Error/Error.vue?vue&type=style&index=0&id=395d98a1&lang=scss&scoped=true&)
Module build failed (from ./node_modules/sass-loader/lib/loader.js):

        background-color: $colorRedNewsletterBtn;
                     ^
      Undefined variable: "$colorRedNewsletterBtn".
      in /Users/cojok/Sites/checkout/src/components/Error/Error.vue (line 23, column 23)
 @ ./src/components/Error/Error.vue?vue&type=style&index=0&id=395d98a1&lang=scss&scoped=true& (./node_modules/style-loader!./node_modules/css-loader!./node_modules/vue-loader/lib/loaders/stylePostLoader.js!./node_modules/sass-loader/lib/loader.js!./node_modules/vue-loader/lib??vue-loader-options!./src/components/Error/Error.vue?vue&type=style&index=0&id=395d98a1&lang=scss&scoped=true&) 2:14-316 21:1-42:3 22:19-321
 @ ./src/components/Error/Error.vue?vue&type=style&index=0&id=395d98a1&lang=scss&scoped=true&
 @ ./src/components/Error/Error.vue
 @ ./src/components/Error/Error.stories.js
 @ ./src/components sync \.stories\.js$
 @ ./.storybook/config.js
 @ multi ./node_modules/@storybook/core/dist/server/common/polyfills.js ./node_modules/@storybook/core/dist/server/preview/globals.js ./.storybook/config.js ./node_modules/webpack-hot-middleware/client.js?reload=true
```

**Expected behavior**
I expected that storybook would render the component as in the normal setup. As i don't need to rewrite the components to manually add the helpers.scss in each one of them.

**Code snippets**
***.storybook/webpack.config.js***
```
const path = require('path')
const alias = {
  '@': path.join(__dirname, 'src'),
  'utils': path.join(__dirname, 'src/_utils'),
  'components': path.join(__dirname, 'src/components')
}
module.exports = async ({ config, mode }) => {
  config.resolve.alias = {
    ...config.resolve.alias,
    ...alias
  }
  config.module.rules.push({
    test: /\.scss$/,
    loaders: ['style-loader', 'css-loader', 'sass-loader'],
    include: path.resolve(__dirname, "../")
  });

  return config
}
```
***.storybook/config.js***
```
import { configure } from '@storybook/vue';
// // automatically import all files ending in *.stories.js
const req = require.context('../src/components', true, /\.stories\.js$/);
function loadStories() {
  req.keys().forEach(filename => req(filename));
}

configure(loadStories, module);
```
***vue.config.js***
```
const path = require('path')
module.exports = {
  css: {
    loaderOptions: {
      // pass options to sass-loader
      sass: {
        // @/ is an alias to src/
        // so this assumes you have a file named `src/variables.scss`
        data: `@import "@/sass/helpers.scss";`
      }
    }
  },
  configureWebpack: {
    resolve: {
      alias: {
        '@': path.join(__dirname, 'src'),
        'utils': path.join(__dirname, 'src/_utils'),
        'components': path.join(__dirname, 'src/components')
      }
    }
  }
}

```
**System:**
 - OS: MacOS 10.13.6 (High Sierra)
 - Device: iMac (Retina 4K, 21.5-inch, 2017)
 - Browser: not relevant
 - Frameworks: 
    @storybook/vue v5.0.11
    vue v2.6.10

Thanks.

**Comment by goldstork** [comment]
Hi!

This is my config and it works for me.
I think this is not the best option, but others did not work for me.

`
const path = require('path')
const rootPath = path.resolve(__dirname, '../src')

module.exports = ({ config }) => {
	config.module.rules.push({
		test: /\.scss$/,
		use: [
			'vue-style-loader',
			'css-loader',
			{
				loader: 'sass-loader',
				options: {
					data: `
						@import "@/scss/common/_functions.scss";
						@import "@/scss/variables/_global.scss";
						@import "@/scss/variables/_media.scss";
						@import "@/scss/common/_mixins.scss";
						@import "@/scss/common/_media.scss";
					`
				}
			}
		],
	})

	config.module.rules.push({
		test: /\.pug$/,
		loader: 'pug-plain-loader'
	})

	config.module.rules.push({
		test: /\.svg$/,
		use: [
			'svg-sprite-loader',
			'svgo-loader'
		]
	})

	config.resolve.alias['@'] = rootPath
	config.resolve.alias['~'] = rootPath

	return config
}`

**Comment by cojok** [comment]
Hi @goldstork,

Thanks a lot for the suggestion. It works!

**Comment by Donghoon759** [comment]
> Hi!
> 
> This is my config and it works for me.
> I think this is not the best option, but others did not work for me.
> 
> **.storybook\webpack.config.js**
> 
> ```
> const path = require('path')
> const rootPath = path.resolve(__dirname, '../src')
> 
> module.exports = ({ config }) => {
> 	config.module.rules.push({
> 		test: /\.scss$/,
> 		use: [
> 			'vue-style-loader',
> 			'css-loader',
> 			{
> 				loader: 'sass-loader',
> 				options: {
> 					data: `
> 						@import "@/scss/common/_functions.scss";
> 						@import "@/scss/variables/_global.scss";
> 						@import "@/scss/variables/_media.scss";
> 						@import "@/scss/common/_mixins.scss";
> 						@import "@/scss/common/_media.scss";
> 					`
> 				}
> 			}
> 		],
> 	})
> 
> 	config.module.rules.push({
> 		test: /\.pug$/,
> 		loader: 'pug-plain-loader'
> 	})
> 
> 	config.module.rules.push({
> 		test: /\.svg$/,
> 		use: [
> 			'svg-sprite-loader',
> 			'svgo-loader'
> 		]
> 	})
> 
> 	config.resolve.alias['@'] = rootPath
> 	config.resolve.alias['~'] = rootPath
> 
> 	return config
> }
> ```

This worked for me. Importing css in configure.js doesn't have effect in my case. Thanks a lot. I spent few hours to solve this issue. As a version increases, code above doesn't work now. So I modified a little bit.

data -> additionalData

{
> 				loader: 'sass-loader',
> 				options: {
> 					additionalData: `
> 						@import "@/scss/common/_functions.scss";
> 						@import "@/scss/variables/_global.scss";
> 						@import "@/scss/variables/_media.scss";
> 						@import "@/scss/common/_mixins.scss";
> 						@import "@/scss/common/_media.scss";
> 					`
> 				}
> 			}

**Comment by davestewart** [comment]
After installing the [scss presets](https://github.com/storybookjs/presets/tree/master/packages/preset-scss) I needed to add global variables and tried the steps above.

Unfortunately, it caused [other errors](https://github.com/vuejs-templates/webpack-simple/issues/107) with loaders being loaded twice (makes sense, as `scss` would be added twice; once in the preset and once here).

A bif of experimentation with the config, and it looks like this can be solved directly in the storybook config file:

```js
module.exports = {
  'stories': [
    '../src/**/*.stories.mdx',
    '../src/**/*.stories.@(js|jsx|ts|tsx)'
  ],
  'addons': [
    '@storybook/addon-links',
    '@storybook/addon-essentials',
    {
      name: '@storybook/preset-scss',
      options: {
        sassLoaderOptions: {
          data: `@import "./src/core/assets/styles/_variables.scss";`,
        },
      }
    }
  ]
}
```

Hope that helps.

**Comment by js-wi5** [comment]
Thanks @davestewart that's much simpler than the other solutions and worked for me too.

It should be noted that if your version of sass-loader is v10.* (as on my project) then the above will work when `data:` is replaced with `additionalData:`.

**Comment by proxycase** [comment]
Thanks to others this is how I solved my situation with vue (2.6.6) / nuxt (2.12.2)

Had to remove the `@storybook/addon-postcss` btw due to conflicts with some existing webpack rules I believe but am not sure. Also explicitly using webpack 4.43.0 in my package, as well as style-loader (2.0.0), sass-loader (7.1.0) and css-loader (4.3.0).

```js
const path = require("path");

module.exports = {
  stories: ["../components/**/*.stories.js"],
  addons: [
    "@storybook/addon-links",
    "@storybook/addon-essentials",
    "storybook-addon-pseudo-states",
    "storybook-addon-outline",
  ],
  webpackFinal: async (config, { configType }) => {
    config.module.rules.push({
      test: /\.scss$/,
      use: [
        "style-loader",
        // Translates CSS into CommonJS
        "css-loader",
        {
          // Compiles Sass to CSS
          loader: "sass-loader",
          options: {
            data: `@import "./assets/scss/app.scss";`,
            includePaths: [__dirname, "./assets/**/*"],
          },
        },
      ],
      resolve: {
        alias: {
          "@": path.resolve(__dirname, "assets"),
          "~": path.resolve(__dirname, "assets"),
        },
      },
    });

    // Return the altered config
    return config;
  },
};
```

**Comment by j-d-carmichael** [comment]
@proxycase  thanks for the solution!

For anyone landing here, the solution did work for me with a slight adjustment (for webpack 4) and my main.js file looks like this:

https://stackoverflow.com/questions/70924456/storybook-vue-sass-additionaldata-not-working/70924623#70924623
